lib/resources/storage_provider.dart

```diff
@@ -1,13 +1,14 @@
-import 'package:cached_network_image/cached_network_image.dart';
-import 'package:firebase_storage/firebase_storage.dart';
+import 'package:acs_upb_mobile/resources/storage/fire_storage_service.dart';
 import 'package:flutter/material.dart';
 
-class StorageProvider {
+class StorageProvider with ChangeNotifier {
   StorageProvider._();
 
-  static Future<ImageProvider<dynamic>> imageFromPath(String path) async {
-    final StorageReference ref = FirebaseStorage.instance.ref().child(path);
-    final String url = await ref.getDownloadURL();
-    return CachedNetworkImageProvider(url);
+  static Future<String> findImageUrl(BuildContext context, String iconPath) async {
```

Comment:
This method adds nothing over `loadImage` in `FireStorageService`. I think you can just rename `FireStorageService` to `StorageProvider` and delete this file.